twitter/tweetwarc.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+"""
+Create WARC files from tweets stored in Apache Kafka in user defined dir.
+Read configuration from YAML file. Run like this:
+    tweetwarc.py -c tweetwarc.yaml -d /dir-to-save-warc
+"""
+from __future__ import unicode_literals
+import argparse
+import yaml
+import hashlib
+import uuid
+import json
+import logging
+import os
+from datetime import datetime
+from time import time
+from hanzo.warctools import WarcRecord
+from hanzo.warctools.warc import warc_datetime_str
+from kafka import KafkaConsumer
+
+logging.basicConfig(level=logging.INFO)
+
+
+def warc_uuid(text):
+    """Utility method for WARC header field urn:uuid"""
+    return ("<urn:uuid:%s>" %
+            uuid.UUID(hashlib.sha1(text).hexdigest()[0:32])).encode('ascii')
+
+
+def warc_filename(directory):
+    """WARC filename example: /tmp/tweets20170307100027.warc.gz.open
+    After the file is closed, remove the .open suffix
+    """
+    return "%s/tweets%s.warc.gz.open" % (
+        directory, datetime.utcnow().strftime('%Y%m%d%H%M%S'))
+
+def warcinfo_record(warc_filename):
+    """Return warcinfo WarcRecord.
+    Required to write in the beginning of a WARC file.
+    """
+    warc_date = warc_datetime_str(datetime.now())
+    metadata = "\r\n".join((
+        "format: WARC File Format 1.0",
+        "conformsTo: http://bibnum.bnf.fr/WARC/WARC_ISO_28500_version1_latestdraft.pdf"
+    ))
+    return WarcRecord(
+        headers=[
+            (WarcRecord.TYPE, WarcRecord.WARCINFO),
+            (WarcRecord.CONTENT_TYPE, b'application/warc-fields'),
+            (WarcRecord.ID, warc_uuid(metadata+warc_date)),
+            (WarcRecord.DATE, warc_date),
+            (WarcRecord.FILENAME, warc_filename)
+        ],
+        content=(b'application/warc-fields', metadata),
```

Comment:
So are you saying content does not need to have CRLF at the end, because `WarcRecord` is writing out two new `nl`s (which has CRLF) at the end of the record (line 125 and 126)?
These two new lines are end-of-record padding prescribed by WARC specification. They are not included in Content-Length. warcinfo record content should have its own CRLF at the end. Please add it to make it right.